tests/integration/lighthouse_api_test.py

```diff
@@ -0,0 +1,63 @@
+import contextlib
+import json
+import os
+import pprint
+from pathlib import Path
+
+import pytest
+from httpx import AsyncClient
+
+from lighthouse.api import Input, Output, app
+
+
+@pytest.fixture()
+async def client() -> AsyncClient:
+    async with AsyncClient(app=app, base_url="http://localhost") as client:
+        yield client
+
+
+@pytest.mark.asyncio
+async def test_ping_endpoint(client):
+    response = await client.get("/_ping")
+    assert response.status_code == 200
+    assert response.json() == {"status": "ok"}
+
+
+@contextlib.contextmanager
+def popen_mock():
+    """Mock the call to asyncio.create_subprocess_exec."""
+
+    def popen(*args, **kwargs):
+        class MockedProcess:
+            def __init__(self, *args, **kwargs):
+                self.returncode = 0
+
+            async def communicate(self) -> tuple[bytes, bytes]:
+                # return the minimal structure required from lighthouse standard output
+                return b'{"categories":{"accessibility":{"score":0.1337}}}', b"stderr mock"
+
+        return MockedProcess(*args, **kwargs)
+
+    from unittest import mock
+    from unittest.mock import AsyncMock
+
+    with mock.patch("asyncio.create_subprocess_exec", new=AsyncMock(side_effect=popen)) as mocked:
+        yield mocked
+
+
+@pytest.mark.asyncio
+async def test_accessibility_endpoint(client):
+    """
+    This test:
+     - does not need lighthouse installed / configured, because it mocks the process.Open call.
+    """
+    input = Input(url="https://google.com", strategy="desktop")
+    with popen_mock():
+        response = await client.post("/accessibility", json=input.dict(), timeout=1)
+
+    pprint.pprint(response.json())
+
+    assert response.status_code == 200
+
+    # make sure, that our result actually complies with the promised open api spec.
+    Output.parse_obj(json.loads(response.text))
```

Comment:
Why is there no assert here? I presume you use an exception instead of an assert here. This seems a bit dirty to me, since the test pipeline will fail if this parsing does not work and we may not get information if other tests fail too, right?

Reply:
Correct, the test essentially checks that there is no exception.
Tests are started independently by pytest (afaik via fork or something similar), hence an exception from one unittest will not prevent running the other tests.

Reply:
Still feels a bit weird, like an unfinished test.